x-pack/plugins/ml/public/cases/index.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { CasesUiSetup } from '@kbn/cases-plugin/public/types';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiButtonIcon, EuiDescriptionList } from '@elastic/eui';
+import type { PersistableStateAttachmentViewProps } from '@kbn/cases-plugin/public/client/attachment_framework/types';
+import type { CoreStart } from '@kbn/core/public';
+import { getAnomalySwimLaneEmbeddableComponent } from '../embeddables/anomaly_swimlane';
+import type { MlStartDependencies } from '../plugin';
+import { PLUGIN_ICON } from '../../common/constants/app';
+import { ANOMALY_SWIMLANE_EMBEDDABLE_TYPE, AnomalySwimlaneEmbeddableInput } from '../embeddables';
+
+const AttachmentActions: React.FC = () => {
+  return (
+    <EuiButtonIcon
+      data-test-subj="test-attachment-action"
+      onClick={() => {}}
+      iconType="boxesHorizontal"
+      aria-label="See attachment"
+    />
+  );
+};
+
+export function registerCasesAttachment(
+  cases: CasesUiSetup,
+  coreStart: CoreStart,
+  pluginStart: MlStartDependencies
+) {
+  cases.attachmentFramework.registerPersistableState({
+    id: ANOMALY_SWIMLANE_EMBEDDABLE_TYPE,
+    icon: PLUGIN_ICON,
+    // TODO check where this name is presented
+    displayName: i18n.translate('xpack.ml.cases.anomalySwimLane.displayName', {
+      defaultMessage: 'Anomaly swim lane',
+    }),
+    getAttachmentViewObject: () => ({
+      event: (
+        <FormattedMessage
+          id="xpack.ml.cases.anomalySwimLane.embeddableAddedEvent"
+          defaultMessage="added the Anomaly Swim Lane embeddable"
+        />
+      ),
+      timelineAvatar: PLUGIN_ICON,
+      actions: <AttachmentActions />,
+      children: React.lazy(() => {
```

Comment:
@jonathan-buttner if a user does not have ML permissions, the comment is shown, with an error callout inside the comment. This behavior is ok I think, but @darnautov and I were wondering if we could or should hide the entire comment in this case?

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/7405507/185906345-838e6cda-9d35-4c5a-a515-bc36b33ce078.png">

Reply:
Yeah that would work. I wonder if it'd be useful to indicate to the user that they don't have permissions to view a portion of the case? I'd create an issue and we can ping Shani.